### PR TITLE
Update Keychron Q3

### PR DESCRIPTION
**Pull request by adophoxia**
## Description
Superseeds https://github.com/qmk/qmk_firmware/pull/21406, but this will only cover one board from each series update of PRs.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/keychron/q3/ansi_encoder/info.json [hunk 1]
```diff
@@ -1,23 +1,112 @@
...
+            {"matrix": [5, 6], "x": 83, "y": 64, "flags": 4},                               
+            {"matrix": [5, 10], "x": 131, "y": 64, "flags": 1}, 
+            {"matrix": [5, 11], "x": 148, "y": 64, "flags": 1}, 
+            {"matrix": [5, 12], "x": 164, "y": 64, "flags": 1}, 
+            {"matrix": [5, 13], "x": 180, "y": 64, "flags": 1}, 
+            {"matrix": [5, 14], "x": 198, "y": 64, "flags": 1}, 
+            {"matrix": [5, 15], "x": 211, "y": 64, "flags": 1}, 
+            {"matrix": [4, 14], "x": 224, "y": 64, "flags": 1}
+        ]
+    },
```
Comment (dunk2k) [inline keyboards/keychron/q3/ansi_encoder/info.json:109]:
```suggestion
    },
    "community_layouts": ["tkl_f13_ansi"],
```

File: keyboards/keychron/q3/ansi/info.json [hunk 2]
```diff
@@ -1,18 +1,108 @@
...
+            {"matrix": [5, 13], "x": 180, "y": 64, "flags": 1}, 
+            {"matrix": [5, 14], "x": 198, "y": 64, "flags": 1}, 
+            {"matrix": [5, 15], "x": 211, "y": 64, "flags": 1}, 
+            {"matrix": [4, 14], "x": 224, "y": 64, "flags": 1}
+        ]
+    },
+    "matrix_pins": {
+        "rows": ["B5", "B4", "B3", "A15", "A14", "A13"],
+        "cols": ["C14", "C15", "A0", "A1", "A2", "A3", "A4", "A5", "A6", "A7", "B0", "B1", "A8", "A9", "A10", "H3"]
     },
```
Comment (dunk2k) [inline keyboards/keychron/q3/ansi/info.json:105]:
```suggestion
    },
    "community_layouts": ["tkl_ansi"],
```

File: keyboards/keychron/q3/iso/info.json [hunk 3]
```diff
@@ -1,18 +1,109 @@
...
+            {"matrix": [5, 13], "x": 180, "y": 64, "flags": 1}, 
+            {"matrix": [5, 14], "x": 198, "y": 64, "flags": 1}, 
+            {"matrix": [5, 15], "x": 211, "y": 64, "flags": 1}, 
+            {"matrix": [4, 14], "x": 224, "y": 64, "flags": 1}
+        ]
+    },
+    "matrix_pins": {
+        "rows": ["B5", "B4", "B3", "A15", "A14", "A13"],
+        "cols": ["C14", "C15", "A0", "A1", "A2", "A3", "A4", "A5", "A6", "A7", "B0", "B1", "A8", "A9", "A10", "H3"]
     },
```
Comment (dunk2k) [inline keyboards/keychron/q3/iso/info.json:106]:
```suggestion
    },
    "community_layouts": ["tkl_iso"],
```

File: keyboards/keychron/q3/iso_encoder/info.json [hunk 4]
```diff
@@ -1,23 +1,113 @@
...
+            {"matrix": [5, 6], "x": 83, "y": 64, "flags": 4},                               
+            {"matrix": [5, 10], "x": 131, "y": 64, "flags": 1}, 
+            {"matrix": [5, 11], "x": 148, "y": 64, "flags": 1}, 
+            {"matrix": [5, 12], "x": 164, "y": 64, "flags": 1}, 
+            {"matrix": [5, 13], "x": 180, "y": 64, "flags": 1}, 
+            {"matrix": [5, 14], "x": 198, "y": 64, "flags": 1}, 
+            {"matrix": [5, 15], "x": 211, "y": 64, "flags": 1}, 
+            {"matrix": [4, 14], "x": 224, "y": 64, "flags": 1}
+        ]
+    },
```
Comment (dunk2k) [inline keyboards/keychron/q3/iso_encoder/info.json:110]:
```suggestion
    },
    "community_layouts": ["tkl_f13_iso"],
```

File: keyboards/keychron/q3/jis/info.json [hunk 5]
```diff
@@ -1,18 +1,112 @@
...
+            {"matrix": [5, 13], "x": 180, "y": 64, "flags": 1}, 
+            {"matrix": [5, 14], "x": 198, "y": 64, "flags": 1}, 
+            {"matrix": [5, 15], "x": 211, "y": 64, "flags": 1}, 
+            {"matrix": [4, 14], "x": 224, "y": 64, "flags": 1}
+        ]
+    },
+    "matrix_pins": {
+        "rows": ["B5", "B4", "B3", "A15", "A14", "A13"],
+        "cols": ["C14", "C15", "A0", "A1", "A2", "A3", "A4", "A5", "A6", "A7", "B0", "B1", "A8", "A9", "A10", "H3"]
     },
```
Comment (dunk2k) [inline keyboards/keychron/q3/jis/info.json:109]:
```suggestion
    },
    "layout_aliases": {
        "LAYOUT_tkl_jis": "LAYOUT_jis_91"
    },
    "community_layouts": ["tkl_jis"],
```

File: keyboards/keychron/q3/jis_encoder/info.json [hunk 6]
```diff
@@ -1,23 +1,117 @@
...
+            {"matrix": [5, 9], "x": 119, "y": 64, "flags": 1},                               
+            {"matrix": [5, 10], "x": 135, "y": 64, "flags": 1}, 
+            {"matrix": [5, 11], "x": 151, "y": 64, "flags": 1}, 
+            {"matrix": [5, 12], "x": 166, "y": 64, "flags": 1}, 
+            {"matrix": [5, 13], "x": 180, "y": 64, "flags": 1}, 
+            {"matrix": [5, 14], "x": 198, "y": 64, "flags": 1}, 
+            {"matrix": [5, 15], "x": 211, "y": 64, "flags": 1}, 
+            {"matrix": [4, 14], "x": 224, "y": 64, "flags": 1}
+        ]
+    },
```
Comment (dunk2k) [inline keyboards/keychron/q3/jis_encoder/info.json:114]:
```suggestion
    },
    "layout_aliases": {
        "LAYOUT_tkl_f13_jis": "LAYOUT_jis_92"
    },
    "community_layouts": ["tkl_f13_jis"],
```

File: keyboards/keychron/q3/readme.md [hunk 7]
```diff
@@ -1,22 +1,30 @@
 # Keychron Q3
 
+![Keychron Q3](https://i.imgur.com/KjP4nqd.jpg)
```
Comment (dunk2k) [inline keyboards/keychron/q3/readme.md:3]:
```suggestion
![Keychron Q3](https://i.imgur.com/KjP4nqdh.jpg)
```

**Review by dunk2k — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]
LGTM.  Though, the community layout support would be good to have, too.

**Review by dunk2k — APPROVED** [review]

File: keyboards/keychron/q3/config.h [hunk 8]
```diff
@@ -43,61 +36,13 @@
 /* Disable DIP switch in matrix data */
 #define MATRIX_MASKED
 
+/* Encoder Configuration*/
+#ifdef ENCODER_ENABLE
+    #define ENCODER_DEFAULT_POS 0x3
+#endif
```
Comment (drashna) [inline keyboards/keychron/q3/config.h:42]:
```suggestion
#define ENCODER_DEFAULT_POS 0x3
```

**Comment by drashna** [comment]
CI failure is due to `brandonschlack` keymap.  Otherwise, LGTM.

**Pull request merged**